File: JAVA/2025-01-08/155652.java
//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
import java.util.*;

public class Main {
    public static void main(String[] args) {
    }
}
    class Solution {
        public String solution(String s, String skip, int index) {
            String answer = "";

            for (int i = 0; i< s.length(); i++)
            {
                char a = s.charAt(i);
                int count = 0;
                int snum = Character.getNumericValue(a);
                for (int j = 0; j < index; j++)
                {
                    a+=1;
                    if ((a) > 'z')
                    {
                        c-=26;
                    }
                    if (skip.contains(String.valueOf(a)))
                    {
                        j--;
                    }
                }
                answer += a;
            }
            return answer;
        }
    }
}

File: C++/2025-01-08/155652.cpp
#include <string>
#include <vector>
#include <unordered_set>
using namespace std;

string solution(string s, string skip, int index) 
{
    string answer = "";
	unordered_set<char> ignore_ch;

	for (int i = 0; i < skip.size(); i++)
	{
		ignore_ch.insert(skip[i]);
	}

	for (int i = 0; i < s.size(); i++)
	{
		char c = s[i];

		for (int j = 0; j < index; j++)
		{
			c = ++c > 'z' ? 'a' : c;

			if (ignore_ch.find(c) != ignore_ch.end())
				j--;
		}
		
		answer += c;
	}

    return answer;
}
